src/service.rs

```diff
@@ -634,31 +579,35 @@ where
                         .await;
                     });
                     self.session_wait_handle
-                        .insert((id, proto_id), (Some(sender), handle));
+                        .insert((id, *proto_id), (Some(sender), handle));
                 }
+            } else {
+                debug!("can't find proto [{}] session handle", proto_id);
             }
         }
     }
 
     fn push_session_message(
-        &mut self,
```

Comment:
It’s a bit simpler leaving `self` and remove two buf parameters from this function.

Reply:
In order to avoid frequent memory allocations, partial borrowing problems are avoided in this way